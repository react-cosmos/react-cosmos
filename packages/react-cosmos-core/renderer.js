module.exports = require('./dist/renderer/index.js');
