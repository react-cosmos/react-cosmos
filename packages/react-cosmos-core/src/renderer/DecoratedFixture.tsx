import React, { useMemo } from 'react';
import { FixtureContext } from '../fixture/FixtureContext.js';
import { FixtureState, SetFixtureState } from '../fixtureState/types.js';
import { getDecoratedFixtureElement } from './getDecoratedFixtureElement/index.js';
import {
  ReactDecorator,
  ReactDecoratorModule,
  ReactFixture,
} from './userModuleTypes.js';

type Props = {
  fixture: ReactFixture;
  systemDecorators: ReactDecorator[];
  userDecoratorModules: ReactDecoratorModule[];
  fixtureState: FixtureState;
  setFixtureState: SetFixtureState;
  renderKey: number;
  onErrorReset?: () => unknown;
};
export function DecoratedFixture({
  fixture,
  systemDecorators,
  userDecoratorModules,
  fixtureState,
  setFixtureState,
  renderKey,
  onErrorReset = noop,
}: Props) {
  // Prevent unintentional renders https://reactjs.org/docs/context.html#caveats
  const contextValue = useMemo(
    () => ({ fixtureState, setFixtureState }),
    [fixtureState, setFixtureState]
  );

  const decoratedFixture = useMemo(() => {
    const decorators = [
      ...systemDecorators,
      ...userDecoratorModules.map(m => m.default),
    ];
    return getDecoratedFixtureElement(fixture, decorators, {
      fixtureState,
      setFixtureState,
      onErrorReset,
    });
  }, [
    fixture,
    fixtureState,
    onErrorReset,
    setFixtureState,
    systemDecorators,
    userDecoratorModules,
  ]);

  return (
    <FixtureContext.Provider
      // renderKey controls whether to reuse previous instances (and
      // transition props) or rebuild render tree from scratch
      key={renderKey}
      value={contextValue}
    >
      {decoratedFixture}
    </FixtureContext.Provider>
  );
}

function noop() {}
