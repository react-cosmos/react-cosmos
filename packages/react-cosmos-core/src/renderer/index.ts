export * from './FixtureLoader/getDecoratedFixtureElement/index.js';
export * from './getFixtureList.js';
export * from './getSortedDecoratorsForFixturePath.js';
export * from './reactTypes.js';
export * from './registerPlaygroundShortcuts.js';
export * from './rendererConfig.js';
export * from './types.js';
