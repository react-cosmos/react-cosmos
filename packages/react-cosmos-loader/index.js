// @flow

exports.mount = require('./src/mount').mount;
exports.createContext = require('./src/create-context').createContext;
exports.Loader = require('./src/components/Loader').Loader;
