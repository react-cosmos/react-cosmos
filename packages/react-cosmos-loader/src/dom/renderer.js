// @flow

import { render, unmountComponentAtNode } from 'react-dom';

import type { LoaderWebOpts } from 'react-cosmos-flow/loader';
import type { Renderer } from 'react-cosmos-flow/context';

export function createDomRenderer(opts?: LoaderWebOpts = {}): Renderer {
  const { containerQuerySelector } = opts;
  const container = getDomContainer(containerQuerySelector);

  return function renderer(element) {
    render(element, container);

    return {
      unmount() {
        unmountComponentAtNode(container);
      },
      toJSON: () => null
    };
  };
}

function getDomContainer(querySelector?: string): HTMLElement {
  if (!querySelector) {
    return createDomContainer();
  }

  const existingContainer = document.querySelector(querySelector);
  if (!existingContainer) {
    console.warn(
      `[Cosmos] Could not find ${querySelector} element. Created fresh DOM container.`
    );
    return createDomContainer();
  }

  return existingContainer;
}

function createDomContainer() {
  const existingNode = document.getElementById('root');
  if (existingNode) {
    return existingNode;
  }

  const newNode = document.createElement('div');
  newNode.setAttribute('id', 'root');
  if (document.body) {
    document.body.appendChild(newNode);
  }

  return newNode;
}
