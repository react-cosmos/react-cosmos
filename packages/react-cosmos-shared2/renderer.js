module.exports = require('./dist/renderer');
