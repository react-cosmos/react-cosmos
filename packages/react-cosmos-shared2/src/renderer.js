// @flow

export const RENDERER_MESSAGE_EVENT_NAME = 'cosmos-cmd';
