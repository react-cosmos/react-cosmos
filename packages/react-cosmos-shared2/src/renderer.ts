import { FixtureState } from './fixtureState';

// FYI: Renderer ids are self assigned in remote environments, so uniqueness
// cannot be established by consensus
export type RendererId = string;

export type FixtureNamesByPath = {
  [fixturePath: string]: null | string[];
};

export type FixtureId = {
  path: string;
  name: null | string;
};

export type PingRenderersRequest = {
  type: 'pingRenderers';
};

export type SelectFixtureRequest = {
  type: 'selectFixture';
  payload: {
    rendererId: RendererId;
    fixtureId: FixtureId;
    fixtureState: null | FixtureState;
  };
};

export type UnselectFixtureRequest = {
  type: 'unselectFixture';
  payload: {
    rendererId: RendererId;
  };
};

export type SetFixtureStateRequest = {
  type: 'setFixtureState';
  payload: {
    rendererId: RendererId;
    // The fixture ID is sent alongside the fixture state change to ensure
    // that the fixture state is only paired with its corresponding fixture
    fixtureId: FixtureId;
    fixtureState: null | FixtureState;
  };
};

export type RendererRequest =
  | PingRenderersRequest
  | SelectFixtureRequest
  | UnselectFixtureRequest
  | SetFixtureStateRequest;

export type OnRendererRequest = (msg: RendererRequest) => unknown;

export type RendererReadyResponse = {
  type: 'rendererReady';
  payload: {
    rendererId: RendererId;
    fixtures: FixtureNamesByPath;
  };
};

export type RendererErrorResponse = {
  type: 'rendererError';
  payload: {
    rendererId: RendererId;
  };
};

export type FixtureListUpdateResponse = {
  type: 'fixtureListUpdate';
  payload: {
    rendererId: RendererId;
    fixtures: FixtureNamesByPath;
  };
};

// Caused by an organic state change inside the renderer. Also dispatched
// after a fixtureSelect request, when rendering stateful components, as their
// initial state is read.
export type FixtureStateChangeResponse = {
  type: 'fixtureStateChange';
  payload: {
    rendererId: RendererId;
    // The fixture ID is sent alongside the fixture state to ensure that the
    // fixture state is only paired with its corresponding fixture
    fixtureId: FixtureId;
    // Entire fixture state is included
    fixtureState: null | FixtureState;
  };
};

export type RendererResponse =
  | RendererReadyResponse
  | RendererErrorResponse
  | FixtureListUpdateResponse
  | FixtureStateChangeResponse;

export type OnRendererResponse = (msg: RendererResponse) => unknown;

export const RENDERER_MESSAGE_EVENT_NAME = 'cosmos-cmd';
