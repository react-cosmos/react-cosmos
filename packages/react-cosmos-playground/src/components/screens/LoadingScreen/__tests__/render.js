import React from 'react';
import { Loader } from 'react-cosmos-loader';
import renderer from 'react-test-renderer';
import defaultFixture from '../__fixtures__/default';

describe('LoadingScreen', () => {
  let component;

  beforeEach(() => {
    component = renderer.create(<Loader fixture={defaultFixture} />);
  });

  it('renders correctly', () => {
    expect(component.toJSON()).toMatchSnapshot();
  });
});
