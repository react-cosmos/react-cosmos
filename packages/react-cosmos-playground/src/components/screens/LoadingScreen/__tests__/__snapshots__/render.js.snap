// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingScreen renders correctly 1`] = `
<div
  className="root"
>
  <div
    className="inner innerDelayed"
  >
    <img
      src={Object {}}
    />
    <div>
      <p
        className="header"
      >
        Loading...
      </p>
      <p>
        <em>
          Everyday I'm bundling, ev-everyday I'm bundling ðŸŽµ
        </em>
      </p>
    </div>
  </div>
</div>
`;
